src/noLifeCycleCallRule.ts

```diff
@@ -0,0 +1,59 @@
+import { LifecycleHooks } from '@angular/compiler/src/lifecycle_reflector';
+
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import { sprintf } from 'sprintf-js';
+import { NgWalker } from './angular/ngWalker';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: 'no-life-cycle-call',
+    type: 'maintainability',
+    description: 'Disallows explicit calls to lifecycle hooks.',
+    rationale: 'Explicit calls to lifecycle hooks is confusing.',
+    options: null,
+    optionsDescription: 'Not configurable.',
+    typescriptOnly: true,
+  };
+
+  static FAILURE_STRING: string = 'Avoid explicitly calls to lifecycle hooks in class "%s"';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    return this.applyWithWalker(new ExpressionCallMetadataWalker(sourceFile, this.getOptions()));
+  }
+}
+
+export class ExpressionCallMetadataWalker extends NgWalker {
+  private readonly lifecycles: (keyof typeof LifecycleHooks)[] = [
+    'AfterContentChecked',
+    'AfterContentInit',
+    'AfterViewChecked',
+    'AfterViewInit',
+    'OnChanges',
+    'DoCheck',
+    'OnDestroy',
+    'OnInit'
+  ];
+
+  visitCallExpression(node: ts.CallExpression): void {
+    this.validateCallExpression(node);
+    super.visitCallExpression(node);
+  }
+
+  private validateCallExpression(node: ts.CallExpression): void {
+    const name = (node.expression as any).name;
+
+    if (!name || this.lifecycles.indexOf(name.text.slice(2)) === -1) {
```

Comment:
This may still return some false positives if the method is called `myOnInit` (not that it's a good idea to name a method this way).

We can introduce a local `LifecycleHooksMethods` union type:

```ts
type LifecycleHooksMethods = 'ngOnInit' | ...;
```

and perform the check with `===` instead of `indexOf`.

Reply:
Good catch! I've fixed it, please take a look.

Reply:
Sure, you can add tests.